Docs updatedocs: improve Local Development guide with clarity, cleanup steps, and example

vannu07 opened the pull request:
This pull request updates the `DEVELOPMENT.md` guide to make it more comprehensive and easier to follow for local development.

The following improvements were made:
- Added a clear step-by-step setup guide for cloning and installing the gcloud-mcp server.
- Provided a breakdown of the `--local` vs. remote flags for initializing the server.
- Included an example walkthrough to better illustrate the local development workflow.
- Added a new "Cleanup / Reset" section to help users unlink their local installation and revert to the remote registry.

This update should help new and existing developers more easily set up and troubleshoot their local development environment.

burkedavison commented:
Hey @vannu07 , Please resolve the conventional commits check so we can continue with this review.

vannu07 commented:
> Hey @vannu07 , Please resolve the conventional commits check so we can continue with this review.

Hey @burkedavison,

I've fixed the conventional commits check! I updated the pull request title to docs: improve local development guide.

The check should now be passing, and you can continue with the review.

burkedavison commented:
Closing this PR due to failing presubmits. Please feel free to reopen if working on resolving.
